### ISSUE TITLE
collectionCode false positivie: GUINEA, single letters ejt-561

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#776]
GUINEA is a false positive collectionCode, but in fact should be part of a country

![image](https://user-images.githubusercontent.com/4609956/66276175-a4d70380-e890-11e9-8be3-3f773ca86e96.png)

there is also more 
![image](https://user-images.githubusercontent.com/4609956/66276185-e667ae80-e890-11e9-8ae3-9b24e02ea0fa.png)

**Comment by gsautter** [comment]
Well, the actual problem with the collection codes in this article is this:
![image](https://user-images.githubusercontent.com/5789714/66282384-791f4200-e8bf-11e9-91ae-9334f7d9d951.png)
The list is just too small a list, and with too few collection codes found in GrBio to be recognized as the actual list of collection codes used throughout the article - there has to be some minimum matching ratios to prevent false positives. On top of that, the first collection code is "cAP", which, unlike many others, doesn't start with a capital letter, making it yet harder to recognize.
That said, the whole process drops to the fallback solution, which is going for anything that looks like an acronym. There is a filter based upon country names, but then, "New Guinea" is not an ISO country name, but rather an island half belonging to Indonesia and half occupied by Papua New Guinea (see https://en.wikipedia.org/wiki/New_Guinea), so "New Guinea" fails to tag as a country name, and thus fails to filter out as a collection code.
Bottom line: this is a combination of a too small list of collection codes in "Materials and Methods" and the use of something in place of a country name that is one at most colloquially. Both the main approach and the safety new of the fallback failing on this one ... hard to handle automatically without some serious over-fitting, and the latter would be bound to wreck some havoc elsewhere sooner or later.

**Comment by myrmoteras** [comment]
a list too small?  Essentially these are all the collections mentioned in the article.

Also, we need to consider that names like cAP are picked up and submitted to GRBIO. This in fact is a topic to be discussed with Tim at BioDivNext, and already ongoing.

**Comment by gsautter** [comment]
What I'm saying is that this list of 4 collection codes, with only one of them in GrBio is too small for statistical detection, just too small a sample.
![image](https://user-images.githubusercontent.com/5789714/66314798-db566200-e914-11e9-8f7a-60c2b1446eab.png)
In that regard, these three paragraphs above would also qualify as a list of potential collection codes, so because they start with capitalized words and are in alphabetic order of the latter. That's why the collection code tagger requires such lists to be larger than 3. On top of that, it requires to be at least half of the potential collection codes in GrBio, for additional safety. Imagine the runaway if some arbitrary set of acronyms gets tagged as collection codes ...
